js/digifinex.js

```diff
@@ -0,0 +1,543 @@
+    }
+
+    async fetchOrder (id, symbol = undefined, params = {}) {
+        let response = await this.privateGetOrderInfo (this.extend ({
+            'order_id': parseInt (id),
+        }, params));
+        return this.parseOrder (response);
+    }
+
+    parseOHLCV (ohlcv, market = undefined, timeframe = '1m', since = undefined, limit = undefined) {
+        return [
+            ohlcv[0],
+            ohlcv[5],
+            ohlcv[3],
+            ohlcv[4],
+            ohlcv[2],
+            ohlcv[1],
+        ];
+    }
+
+    async fetchOHLCV (symbol, timeframe = '1m', since = undefined, limit = undefined, params = {}) {
+        let request = {
+            'symbol': symbol,
+            'type': 'kline_' + timeframe,
+        };
+        let response = await this.privateGetKline (this.extend (request, params));
+        this.checkResponse (response, 'fetchOHLCV');
+        return this.parseOHLCVs (response['data'], undefined, timeframe, since, limit);
+    }
+
+    nonce () {
+        return this.milliseconds ();
+    }
+
+    sign (path, api = 'public', method = 'GET', params = {}, headers = undefined, body = undefined) {
+        params['apiKey'] = this.apiKey;
+        if (api === 'private') {
+            this.checkRequiredCredentials ();
+            params['timestamp'] = parseInt (this.seconds ());
+            params['apiSecret'] = this.secret;
+            params = this.keysort (params);
+            let keys = Object.keys (params);
+            let arr = [];
+            for (let i = 0; i < keys.length; i++) {
+                let key = keys[i];
+                let s = params[key].toString ();
+                arr.push (s);
+            }
+            let s = arr.join ('');
+            let sign = this.hash (this.encode (s), 'md5');
+            params['sign'] = sign;
+        }
+        let url = this.urls['api'] + '/' + path;
+        // log.info (url, params);
+        if (method === 'GET') {
+            if (Object.keys (params).length) {
+                url += '?' + this.urlencode (params);
+            }
+        } else if (method === 'POST') {
+            if (Object.keys (params).length) {
+                body = this.urlencode (params);
+            }
+        }
+        let result = { 'url': url, 'method': method, 'body': body, 'headers': headers };
+        // log.info (url);
+        // log.info (result);
+        return result;
+    }
+
+    dateUTC8 (timestampMS) {
+        const timedelta = this.safeValue (this.options, 'timedelta', 8 * 60 * 60 * 1000); // eight hours
+        return this.ymd (timestampMS + timedelta);
+    }
+
+    checkResponse (response, caller) {
+        let code = response['code'];
+        if (code === 0)
+            return;
+        caller = '[' + caller + '] ';
+        if (code === undefined)
+            throw new BadResponse (caller + 'bad response: ' + response);
+        this.checkCode (code, caller);
+    }
+
+    checkCode (code, caller) {
+        if (code === 10001) {
```

Comment:
I suggest that we do the error handling in a unified way, as done in for, example:
- [liqui](https://github.com/ccxt/ccxt/blob/master/js/liqui.js#L76) + [`handleErrors()`](https://github.com/ccxt/ccxt/blob/master/js/liqui.js#L801)
- [bitfinex](https://github.com/ccxt/ccxt/blob/master/js/bitfinex.js#L283) + [`handleErrors()`](https://github.com/ccxt/ccxt/blob/master/js/bitfinex.js#L1041)